flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -317,6 +319,10 @@ public AbstractYarnClusterDescriptor createDescriptor(String defaultApplicationN
 		if (cmd.hasOption(SLOTS.getOpt())) {
 			int slots = Integer.valueOf(cmd.getOptionValue(SLOTS.getOpt()));
 			yarnClusterDescriptor.setTaskManagerSlots(slots);
+		} else if (config.containsKey(ConfigConstants.YARN_VCORES)) {
```

Comment:
Actually, I'm not so sure whether whether we should use the `YARN_VCORES` value as a fallback value for the number of task manager slots. We had this in the past but we changed it so that the default behaviour is if the number of slots is not specified, then it is `1`. Changing this here for Yarn would make the behaviour inconsistent wrt Mesos, for example. From a user's perspective the old behaviour is imo more predictable. However, we should definitely respect the configuration file settings for the task manager slots.